src/aura/HH_ContactCard/HH_ContactCardController.js

```diff
@@ -6,6 +6,7 @@
 
     // the remove button has been pressed, notify our container
     removeContact : function(component, event, helper) {
+        event.preventDefault();
```

Comment:
The button by default wants to submit the form so we are telling it to prevent that default behavior.